distributed/worker.py

```diff
@@ -2858,14 +2866,16 @@ def select_keys_for_gather(self, worker, dep):
         deps = {dep}
 
         total_bytes = self.tasks[dep].get_nbytes()
-        L = self.pending_data_per_worker[worker]
+        tasks = self.data_needed_per_worker[worker]
 
-        while L:
-            ts = L.pop()
+        while tasks:
+            ts = tasks.peek()
```

Comment:
Fix bug where a task that would make an iteration of ``select_keys_for_gather`` exceed total_bytes would not be picked up at the next iteration